Tests/Tests/STPPaymentConfigurationTest.m

```diff
@@ -0,0 +1,120 @@
+}
+
+- (void)testInit {
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+
+    XCTAssertNil(paymentConfiguration.publishableKey);
+    XCTAssertEqual(paymentConfiguration.additionalPaymentMethods, STPPaymentMethodTypeAll);
+    XCTAssertEqual(paymentConfiguration.requiredBillingAddressFields, STPBillingAddressFieldsNone);
+    XCTAssertEqual(paymentConfiguration.requiredShippingAddressFields, PKAddressFieldNone);
+    XCTAssert(paymentConfiguration.verifyPrefilledShippingAddress);
+    XCTAssertEqual(paymentConfiguration.shippingType, STPShippingTypeShipping);
+    XCTAssertNil(paymentConfiguration.companyName); // Defaults to nil for test target
+    XCTAssertNil(paymentConfiguration.appleMerchantIdentifier);
+    XCTAssert(paymentConfiguration.canDeletePaymentMethods);
+}
+
+- (void)testApplePayEnabledSatisfied {
+    id stripeMock = OCMClassMock([Stripe class]);
+    OCMStub([stripeMock deviceSupportsApplePay]).andReturn(YES);
+
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    paymentConfiguration.appleMerchantIdentifier = @"appleMerchantIdentifier";
+    paymentConfiguration.additionalPaymentMethods = STPPaymentMethodTypeAll;
+
+    XCTAssert([paymentConfiguration applePayEnabled]);
+}
+
+- (void)testApplePayEnabledMissingAppleMerchantIdentifier {
+    id stripeMock = OCMClassMock([Stripe class]);
+    OCMStub([stripeMock deviceSupportsApplePay]).andReturn(YES);
+
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    paymentConfiguration.appleMerchantIdentifier = nil;
+    paymentConfiguration.additionalPaymentMethods = STPPaymentMethodTypeAll;
+
+    XCTAssertFalse([paymentConfiguration applePayEnabled]);
+}
+
+- (void)testApplePayEnabledDisallowAdditionalPaymentMethods {
+    id stripeMock = OCMClassMock([Stripe class]);
+    OCMStub([stripeMock deviceSupportsApplePay]).andReturn(YES);
+
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    paymentConfiguration.appleMerchantIdentifier = @"appleMerchantIdentifier";
+    paymentConfiguration.additionalPaymentMethods = STPPaymentMethodTypeNone;
+
+    XCTAssertFalse([paymentConfiguration applePayEnabled]);
+}
+
+- (void)testApplePayEnabledMisisngDeviceSupport {
+    id stripeMock = OCMClassMock([Stripe class]);
+    OCMStub([stripeMock deviceSupportsApplePay]).andReturn(NO);
+
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    paymentConfiguration.appleMerchantIdentifier = @"appleMerchantIdentifier";
+    paymentConfiguration.additionalPaymentMethods = STPPaymentMethodTypeAll;
+
+    XCTAssertFalse([paymentConfiguration applePayEnabled]);
+}
+
+#pragma mark - Description
+
+- (void)testDescription {
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    XCTAssert(paymentConfiguration.description);
```

Comment:
I kind of worry about this implementation drifting, since we add to PaymentConfiguration fairly regularly and it'd be easy to forget to update description. Maybe we can assert that all the properties are present in the string? 

💭  Maybe we should use `class_copyPropertyList` to do this for all `description` implementation tests?